src/programs/someone-tag.ts

```diff
@@ -8,90 +16,64 @@ const QUESTION_SHEET_ID: string =
   "1eve4McRxECmH4dLWLJvHLr9fErBWcCGiH94ihBNzK_s";
 
 @Command({
-  event: DiscordEvent.MESSAGE,
-  trigger: "@someone",
-  allowedRoles: ["Seek Discomfort"],
-  channelNames: ["chat", "chat-too", "4th-chat", "chat-v"],
-  description: "This handler is for the someone command.",
+  event: DiscordEvent.SLASH_COMMAND,
+  root: "someone",
+  description: "Ping a random memeber.",
+  options: [
+    {
+      type: ApplicationCommandOptionType.Boolean,
+      name: "Online",
+      description: "Should the person be Online.",
+      required: true,
+    },
+  ],
 })
-class SomeoneTag implements CommandHandler<DiscordEvent.MESSAGE> {
-  async handle(message: Message): Promise<void> {
-    const allow = await isAllowed(message.author);
+class SomeoneTag implements CommandHandler<DiscordEvent.SLASH_COMMAND> {
+  async handle(interaction: ChatInputCommandInteraction): Promise<void> {
+    const allow = await isAllowed(interaction.user);
 
     if (!allow) {
-      await Tools.handleUserError(
-        message,
-        "You have already used this command today!"
-      );
+      interaction.reply("You have already used this command today!");
       return;
     }
 
-    const words = Tools.stringToWords(message.cleanContent);
-    const arg = words[1];
-    if (arg && arg !== "online")
-      await message.channel.send({
-        content: `Unknown argument "${arg}". Did you mean "online"?`,
-        allowedMentions: { parse: [] },
-      });
-    else {
-      const { member } = message;
-      if (!member) return;
-
-      const target = await getTarget(arg, message);
-      const question = await getQuestion();
-      if (!target) {
-        await message.reply(
-          "There were no available users to ping! This is embarrassing. How could this have happened? There's so many people on here that statistically this message should never even show up. Oh well. Congratulations, I guess."
-        );
-      } else {
-        await updateLastMessage(message);
-        await sendMessage(
-          member,
-          target,
-          question,
-          message.channel as TextChannel
-        );
-      }
-    }
+    const { member } = interaction;
+    if (!member) return;
 
-    await message.delete();
+    const target = await getTarget(
+      interaction.options.getBoolean("Online"),
```

Comment:
Consequence from the change in the options above.

```suggestion
      interaction.options.getBoolean("online"),
```